### DISCUSSION TITLE
Default 'main' git branch missing from added domain - Vercel ENV showing preview deploying to main branch

### DISCUSSION THREAD

**Discussion by jamezening** [vercel/community#2417] [Help] [answered]
### Problem Description

Hi,

I am working on a Vercel project and when deploying to main
VERCEL related variables are showing as `preview`.

```
process.env.NEXT_PUBLIC_VERCEL_ENV: preview
process.env.VERCEL_ENV: preview
```

When trying to set this domain as the main git branch domain I get an error that pops up,
 but couldn't find any useful info in the link provided.

Error:Cannot set Production Branch "main" for a Preview Domain. [Learn More](https://vercel.link/production-branch-as-preview-branch)

I noticed that our Vercel domain is missing a "default" main git branch domain.
![image](https://user-images.githubusercontent.com/6829451/237052018-115b3cfc-57fa-425c-ac7d-d8633d152d7a.png)

---

Another project I am working has a greyed-out, non de-selectable field for configuring the "default" main git branch .
![image](https://user-images.githubusercontent.com/6829451/237055119-412bf4a8-5488-4c00-8d84-2d44c03060d7.png)

---

I am guessing the lack of a "default" main git branch . is causing VERCEL related variables to show as `preview` when our main branch is deployed.

Is there a way I can configure a domain as the "default" main git branch domain?

**Comment by jamezening** [comment]
Manually "Promoting to production" fixes the Vercel ENV to be `production` when deployed, but I am assuming that when deployed via a GitHub PR merge, the same issue will occur.

Any tips on how to ensure that a domain is set to production?
Because our Vercel dashboard is showing that our domain is production, but the behavior is that of a `preview` domain as shown in the first discussion post.

---
<img width="233" alt="image" src="https://user-images.githubusercontent.com/6829451/237063117-3d8fe8e2-f5cf-4920-90bd-24134f7eb378.png">

**Comment by amyegan** [comment]
Sorry you're having domain configuration problems, @vanilla-james. If you can share the domain here, I can look into it

**Reply by jamezening** [reply]
@amyegan 
Thanks for your reply 

Would it be possible to share our Vercel project instead of domain as we are still in private alpha?
https://vercel.com/kryptonium/beoble-toolkit 

We are having trouble with the production url that has been set.

**Reply by jamezening** [reply]
@amyegan 
Did we have an issue with our domain configuration?

Thank you

**Reply by amyegan** [reply, marked as answer]
Sorry for the delay. I don't see any problem with your domain configuration. The custom domain is associated with your most recent production deployment. 

For any other production domains you want to add, you can leave the Git Branch field empty and the domain will automatically be assigned to production. A specific git branch only needs to be set if you want to use a custom domain for a preview branch.

**Reply by jamezening** [reply]
@amyegan
thanks for the confirmation, but the only ways the production deployment is working properly are:
1. deploying from cli with `vercel --prod`
2. clicking 'Promote to production' from our Vercel project dashboard

merging to our production branch (which is `main`) is showing the env as `preview`

any suggestions?
should we remove the domains and add them back?

**Reply by daniel-bergmann** [reply]
Solved it for me
